src/shared/schemas.ts

```diff
@@ -85,15 +88,33 @@ export class SchemaService {
     }
 
     /**
-     * Registers a schema mapping in the schema service.
+     * Registers a schema mapping in the schema service. If the incoming mapping has an owner, then the mapping will be considered "owned".
+     * If the URI is owned (in the owners map), the update will only be processed if the incoming mapping owner is the same as the owned user OR
+     * the type of the schema changed
      *
      * @param mapping
      * @param flush Flush immediately instead of waiting for timer.
      */
     public registerMapping(mapping: SchemaMapping, flush?: boolean): void {
-        this.updateQueue.push(mapping)
-        if (flush === true) {
-            this.processUpdates()
+        // The owner is undefined and needs to be set
+        const isOwnerUndefined =
+            !this.owned.has(mapping.uri) && mapping.owner !== undefined && mapping.schema !== undefined
```

Comment:
nit: "positive logic" is usually easier to read and leads to better names. `isOwned` would be the positive logic form of this, instead of `isOwnerUndefined` (which is equivalent to negative logic "is not owned").